### PR TITLE
fix: horizon sync

### PR DESCRIPTION
**Pull request by SWvheerden**
Description
---
Fixes horizon sync
Run optimized needs to be forced with the correct mutability to ensure it runs from the correct optimized bitmap

Motivation and Context
---
MMR root failure occurs during sync as the optimized run fails in Merkle tree, this provides an already optimized bitmap to the merkle tree.

How Has This Been Tested?
---
Manual sync

### CODE REVIEW

File: base_layer/mmr/src/mutable_mmr.rs [hunk 1]
```diff
@@ -209,13 +210,13 @@ where
     fn get_sub_bitmap(&self, leaf_index: LeafIndex, count: usize) -> Result<Bitmap, MerkleMountainRangeError> {
         let mut deleted = self.deleted.clone();
         if leaf_index.0 > 0 {
-            deleted.remove_range_closed(0..u32::try_from(leaf_index.0 - 1).unwrap())
+            deleted.remove_range(0..u32::try_from(leaf_index.0 - 1).unwrap())
```
Comment (stringhandler) [inline base_layer/mmr/src/mutable_mmr.rs]: Can we check this unwrap and the arithmetic here?

File: base_layer/mmr/src/mutable_mmr.rs [hunk 2]
```diff
@@ -209,13 +210,13 @@ where
...
         }
         let leaf_count = self.mmr.get_leaf_count()?;
         if leaf_count > 1 {
             let last_index = leaf_index.0 + count - 1;
             if last_index < leaf_count - 1 {
-                deleted.remove_range_closed(u32::try_from(last_index + 1).unwrap()..u32::try_from(leaf_count).unwrap());
+                deleted.remove_range(u32::try_from(last_index + 1).unwrap()..u32::try_from(leaf_count).unwrap());
```
Comment (stringhandler) [inline base_layer/mmr/src/mutable_mmr.rs]: Same

File: base_layer/mmr/src/mutable_mmr.rs (hunk 1 above)
Comment (SWvheerden) [reply base_layer/mmr/src/mutable_mmr.rs]: good check, I just rename the api fix

File: base_layer/mmr/src/mutable_mmr.rs [hunk 3]
```diff
@@ -82,8 +83,8 @@ where
     /// nodes in the MMR, while this function returns the number of leaf nodes minus the number of nodes marked for
     /// deletion.
     #[allow(clippy::len_without_is_empty)]
-    pub fn len(&self) -> u32 {
-        self.size - u32::try_from(self.deleted.cardinality()).unwrap()
+    pub fn len(&self) -> Result<u32,MerkleMountainRangeError> {
+        Ok(self.size - u32::try_from(self.deleted.cardinality()).map_err(|_|MerkleMountainRangeError::InvalidMmrSize)?)
```
Comment (stringhandler) [inline base_layer/mmr/src/mutable_mmr.rs]: Check this arithmetic for underflow

**Review by stringhandler — APPROVED** [review]

**Pull request merged**